OpenRobertaServer/staticResources/js/app/simulation/simulationLogic/robot.ev3.js

```diff
@@ -284,62 +284,64 @@ define(['simulation.simulation', 'interpreter.constants', 'simulation.robot', 'g
                 }
                 utterThis.pitch = pitch;
                 utterThis.rate = speed;
+                utterThis.volume = volume;
                 utterThis.onend = function(event) {
                     that.sayText.finished = true;
                 };
                 SpeechSynthesis.speak(utterThis);
-            },
+                //does not work for volume = 0 thus workaround with if statement and cancel
+                if(volume == 0){
+                SpeechSynthesis.cancel();
+                }
```

Comment:
Maybe a better workaround would be to not even start speaking. Otherwise I myself hear a short sound just before it's canceled.

Reply:
I tried that originally but it caused the simulation to not stop. Maybe both is required.

Reply:
Hm okay, this is probably because of the the `onend` callback. A possible workaround would be to have a `setTimout` with `that.sayText.finished = true`. But you can decide this.